src/test/java/dev/openfeature/sdk/providers/memory/InMemoryProviderTest.java

```diff
@@ -125,4 +118,15 @@ void emitChangedFlagsOnlyIfThereAreChangedFlags() {
         await().untilAsserted(() -> verify(handler, times(1))
                 .accept(argThat(details -> details.getFlagsChanged().size() == buildFlags().size())));
     }
+
+    @SneakyThrows
+    @Test
+    void shouldThrowGeneralErrorAfterEmittingErrorEvent() {
+        InMemoryProvider inMemoryProvider = new InMemoryProvider(new HashMap<>());
+        inMemoryProvider.initialize(null);
+        inMemoryProvider.emitProviderError(ProviderEventDetails.builder().build());
+
+        // ErrorCode.GENERAL should be returned when evaluated via the client
```

Comment:
why would it return error after provider error event ?
any reference to this behavior ?